src/app/home/home.component.ts

```diff
@@ -13,17 +14,57 @@ import { EmployeesService } from '../services/employees/employees.service'
 export class HomeComponent implements OnInit {
   employees: IEmployee[] = []
   employeesSubscription: Subscription
+  tableColumns: string[] = ['name', 'birthDate', 'email', 'phoneNumber']
+
+  dataSource: MatTableDataSource<IEmployee>
+
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator
+  @ViewChild(MatSort, { static: false }) sort: MatSort
+
   constructor(private httpClient: HttpClient, private employeesData: EmployeesService) {
     this.employeesData.fetch()
     this.employeesSubscription = this.employeesData.list.subscribe(data => {
       if (data) {
+        console.log(data)
         this.employees = data
       } else {
         this.employees = []
       }
     })
+    this.dataSource = new MatTableDataSource(this.employees)
+  }
+  ngOnInit() {
+    this.dataSource.filterPredicate = (employee: IEmployee, filter: string) => {
+      console.log(employee)
+      console.log(filter)
+      return employee.bio.firstName.toLowerCase() == filter
```

Comment:
This probably should be `return employee.bio.firstName.toLowerCase().includes(filter.toLowerCase())`

That way, it's totally case insensitive and works for sub-strings (jo yields John and Johanssen, or Joan, etc).

We may want to also extend it to last name as well...
`return employee.bio.firstName.toLowerCase().includes(filter.toLowerCase()) || employee.bio.lastName.toLowerCase().includes(filter.toLowerCase())`

Reply:
I can't tag it directly, but the `unsubscribe` in `ngOnDestroy` needs parens.  `unsubscribe()`

Reply:
fixed @brendoncaulkins good catches!